src/bit-systems/audio-zone-system.ts

```diff
@@ -0,0 +1,318 @@
+import { defineQuery, enterQuery, exitQuery } from "bitecs";
+import { getScene, HubsWorld } from "../app";
+import { AudioEmitter, AudioZone, AudioListenerTag } from "../bit-components";
+import { Box3, BoxGeometry, DoubleSide, MeshBasicMaterial, Object3D, Ray, Vector3, Mesh, BoxHelper } from "three";
+import { AUDIO_ZONE_FLAGS } from "../inflators/audio-zone";
+import { disposeMaterial, disposeNode } from "../utils/three-utils";
+import { AudioSettings } from "../components/audio-params";
+import { updateAudioSettings } from "../update-audio-settings";
+
+const debugObjects = new Map<number, Object3D>();
+
+const DEBUG_BBAA_COLOR = 0x49ef4;
+
+let debugMaterial: MeshBasicMaterial | undefined | null;
+
+const createMaterial = () => {
+  if (!debugMaterial) {
+    debugMaterial = new MeshBasicMaterial({
+      color: DEBUG_BBAA_COLOR,
+      transparent: true,
+      opacity: 0.25,
+      side: DoubleSide
+    });
+  }
+};
+
+getScene().then(() => {
+  (APP.store as any).addEventListener("statechanged", () => {
+    const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+    if (isDebugEnabled) createMaterial();
+    defineQuery([AudioZone])(APP.world).forEach(zone => {
+      if (isDebugEnabled) {
+        !debugObjects.has(zone) && addZoneDebugObject(APP.world, zone);
+      } else {
+        debugObjects.has(zone) && releaseZoneDebugObject(APP.world, zone);
+      }
+    });
+    if (!isDebugEnabled && debugMaterial) {
+      disposeMaterial(debugMaterial);
+      debugMaterial = null;
+    }
+  });
+  const isDebugEnabled = APP.store.state.preferences.showAudioDebugPanel;
+  if (isDebugEnabled) createMaterial();
+});
```

Comment:
I think this block can be moved to a regular system. It doesn't need the to use `getScene` or listen to `statechange`s from the store:

```ts
 let isShowing = false;
 function audioDebugPanelToggleSystem(world: HubsWorld) {
   const shouldShow = APP.store.state.preferences.showAudioDebugPanel;
   if (isShowing && !shouldShow) {
     // teardown
   } else if (!isShowing && shouldShow){
     // setup
   }
   isShowing = shouldShow;
 }
 ```

Reply:
IIUC this would make us repeat this same code in any system where we want to read this preferences parameter (ie. we also need to do this in the audio debug system) or centralize it in a system that calls functions on other systems explicitly.

An alternative to this would be to move preferences to bitecs where we have a `[Xyz]Preference` components, then any system interested could just query enter/exit on the prefference they are monitoring and then access it through the store so you only query when it has actually changed.